src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -211,19 +211,19 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('fragment_renderer.php');
         $loader->load('error_renderer.php');
 
-        if (interface_exists(PsrEventDispatcherInterface::class)) {
+        if ($container->isAvailable('psr/event-dispatcher', PsrEventDispatcherInterface::class)) {
             $container->setAlias(PsrEventDispatcherInterface::class, 'event_dispatcher');
         }
 
         $container->registerAliasForArgument('parameter_bag', PsrContainerInterface::class);
 
-        if (class_exists(Application::class)) {
+        if ($container->isAvailable('symfony/console', Application::class)) {
             $loader->load('console.php');
 
-            if (!class_exists(BaseXliffLintCommand::class)) {
+            if (!$container->isAvailable('symfony/translation', BaseXliffLintCommand::class)) {
                 $container->removeDefinition('console.command.xliff_lint');
             }
-            if (!class_exists(BaseYamlLintCommand::class)) {
+            if (!$container->isAvailable('symfony/yaml', BaseYamlLintCommand::class)) {
```

Comment:
to me, the commands should be registered if installed so that they are usable (especially for such developer-focused commands